http-server/src/main/java/org/triplea/server/user/account/login/authorizer/temp/password/TempPasswordLogin.java

```diff
@@ -0,0 +1,35 @@
+package org.triplea.server.user.account.login.authorizer.temp.password;
+
+import com.google.common.base.Preconditions;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+import javax.annotation.Nonnull;
+import lombok.Builder;
+import org.triplea.http.client.lobby.login.LoginRequest;
+import org.triplea.lobby.server.db.dao.TempPasswordDao;
+
+@Builder
+public class TempPasswordLogin implements Predicate<LoginRequest> {
+
+  @Nonnull private final TempPasswordDao tempPasswordDao;
+  @Nonnull private final BiPredicate<String, String> passwordChecker;
+
+  @Override
+  public boolean test(final LoginRequest loginRequest) {
+    Preconditions.checkNotNull(loginRequest);
+    Preconditions.checkNotNull(loginRequest.getName());
+    Preconditions.checkNotNull(loginRequest.getPassword());
+    return tempPasswordDao
+        .fetchTempPassword(loginRequest.getName())
+        .map(
+            tempPassword -> {
+              if (passwordChecker.test(loginRequest.getPassword(), tempPassword)) {
+                tempPasswordDao.invalidateTempPasswords(loginRequest.getName());
+                return true;
+              } else {
+                return false;
+              }
```

Comment:
The else case is unecessary at this point

Reply:
True, we should perhaps revisit this as a convention.

1) the block structure is easier to grok when there is an explicit else, the return statements line up.
2) Arguably it's easier to understand, when you see if/else block you know that the method terminates in the conditionals and not possibly in some and not others.
3) Perhaps most compelling to me, I had a bug where in one case I missed the 'return' in the 'if' block, the code fell through to the final return and returned the wrong value, eg:

```
if( condition ) {
  // this return statement was missing
}
return false;
```

On the other hand, written as an `if/else` block, the above would have given a compilation error.

So while the `} else {` is unnecessary, removing it is maybe not the best, less I'm not sure is more in this case.